design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,711 @@
+virt-controller monitors the virt-launcher Pod [[1]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-controller/watch/vmi.go#L974).
+
+1. On virt-controller reconcile loop
+- When VMI phase is scheduling and its virt-launcher Pod is ready [[2]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-controller/watch/vmi.go#L509),
+given the network-status annotation exists copy it from the Pod metadata
+to the VMI metadata.annotations.
+
+- When the VMI phase is Running [[3]](https://github.com/kubevirt/kubevirt/blob/1a8f08103e48c5f2bb2f5826d118507ce7ec1f0c/pkg/virt-controller/watch/vmi.go#L576).
+Compare the Pod and VMI networks-status annotation:
+If it's not exists add it to the VMI.
+If they are not the equal, copy the new annotation from the Pod to the VMI.
+
+> **_Note_**: When a VM is migrated successfully virt-controller will pick the new target pod as part of its reconcile loop, which will have the annotation with the new details.
+The new annotation will be copied to the VMI.
+
+2. As part of virt-launcher VMI spec conversion to Libvirt domain:
+- Given the networks-status annotation exists on the VMI object, extract the mapping between SR-IOV networks and the VFs PCI addresses and create each HostDevice accordingly.
+- In case networks-status annotation doesn't exist, fall back to the legacy mapping method (based on environment variables).
+
+### Backward compatibility
+|                                                       | old virt-controller                                | new virt-controller                                                                                                                                                                                                              |
+|:------------------------------------------------------|:---------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Already existing VMs<br/>(old virt-launcher)          | 1. No regression <br/> VFs mapping might be wrong. | 2. No regression <br/><br/>VFs mapping might be wrong.<br/>virt-controller annotates the VMI with the network status annotation from the pod,but virt-launcher will not use it.                                                  |
+| VM creation/restart/migration<br/>(new virt-launcher) | -                                                  | 4. No regression<br/>Once the VM is restarted, VMI or Pod is re-created, the new SR-IOV interfaces assignment logic will apply.<br/>In case the VM is migrated, the VM on target will have SR-IOV interfaces assigned correctly. |
+
+#### Conclusion
+- (2) virt-controller will add the network-status to the VMI, the guest will have no disruption and its VFs mapping will not change.
```

Comment:
how about this?
```suggestion
- (related to solution section 2) virt-controller will add the network-status to the VMI, the guest will have no disruption and its VFs mapping will not change.
```

Reply:
Isn't it clear from the numbering to which cell each bullet is related in the table?
I am asking because I think it's cleaner with only the numbering.